Monika After Story/game/definitions.rpy

```diff
@@ -972,14 +1032,35 @@ python early:
             """
             self.flags &= ~flags
 
+        @classmethod
+        def validateConditionals(cls):
+            """
+            A method to validate conditionals
+
+            ASSUMES:
+                mas_all_ev_db
+            """
+            for ev in mas_all_ev_db.itervalues():
+                if ev.conditional is not None:
+                    try:
+                        eval(cls._conditional_cache[ev.conditional], globals=renpy.store.__dict__, locals=renpy.store.__dict__)
+
+                    except Exception as e:
+                        raise EventException(
+                            "Failed to evaluate the '{0}' conditional for the event with the '{1}' label:\n{2}.".format(
+                                ev.conditional,
+                                ev.eventlabel,
+                                e
```

Comment:
for a better formatted exception, use the `traceback` module (`traceback.format_exc()`). See `zz_spritejsons` for examples.

Reply:
I tried that, and it didn't look informative to me. It just printed more lines which I didn't need (RenPy already prints stack quite nicely), I still looked at the exception in the end. But I'll change that so you can see which variant works better.